datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/cliUtils/DataframeOptions.scala

```diff
@@ -30,20 +32,51 @@ case class DataframeOptions(format: String, options: Map[String, String], path:
     if (format == "jdbc") dfReader.load()
     else dfReader.load(path)
 
-  private def save[T](dfWriter: DataFrameWriter[T], extraPath: String = ""): Unit =
+  private def save[T](dfWriter: DataFrameWriter[T], endPath: String = ""): Unit =
     if (format == "jdbc") dfWriter.save()
-    else dfWriter.save(s"$path$extraPath")
+    else dfWriter.save(endPath)
+
+  def getUniqueFilePath(extraPath: String, conf: Configuration): String = {
+    val fs = FileSystem.get(conf)
+    val basePath = s"$path$extraPath"
+
+    @scala.annotation.tailrec
+    def appendNumberAndTest(namePt1: String,
+                            namePt2: String,
+                            condition: String => Boolean,
+                            count: Int = 1): String = {
+      val newName = s"${namePt1}_run$count$namePt2"
+      if (condition(newName)) { appendNumberAndTest(namePt1, namePt2, condition, count + 1) }
+      else newName
+    }
+
+    if (fs.exists(new Path(basePath))) {
+      appendNumberAndTest(path, extraPath, { x: String => fs.exists(new Path(x)) })
+    } else {
+      basePath
+    }
+  }
 
   def loadDataFrame(implicit spark: SparkSession): DataFrame = {
     val dfReader = spark.read.format(format)
     val withOptions = setOptions(dfReader)
     load(withOptions)
   }
 
-  def writeDataFrame(df: DataFrame, extraPath: String = "")(implicit spark: SparkSession): Unit = {
+  def writeDataFrame(df: DataFrame, extraPath: String = "")
+                    (implicit spark: SparkSession): Unit = {
+    val dfWriter =  df.write.format(format)
+    val withOptions = setOptions(dfWriter)
+    save(withOptions, s"$path$extraPath")
+  }
+
+  def writeNextDataFrame(df: DataFrame, extraPath: String = "")
+                        (implicit spark: SparkSession): String = {
+    val nonExistingPath: String = getUniqueFilePath(extraPath, spark.sparkContext.hadoopConfiguration)
```

Comment:
Why not to call this `uniquePath`/`uniqueFilePath`?

Reply:
used `uniqueFilePath`